File: apps/mobile/src/services/contract.ts
import { api } from './api';
import { ENDPOINTS } from '../constants/api';

export interface Contract {
  id: string;
  chatRoomId: string;
  userAId: string;
  userBId: string;
  status: 'DRAFT' | 'SIGNED';
  contractData: Record<string, any>;
  signatureA: boolean;
  signatureB: boolean;
  signedAt: string | null;
}

export async function getContractByChatRoom(chatRoomId: string): Promise<Contract> {
  return api.get<Contract>(ENDPOINTS.CONTRACTS.BY_CHAT_ROOM(chatRoomId));
}

export async function initContract(chatRoomId: string): Promise<Contract> {
  return api.post<Contract>(ENDPOINTS.CONTRACTS.INIT, { chatRoomId });
}

File: apps/api/app/schemas/contract.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class ContractStatus(str, Enum):
    DRAFT = "DRAFT"
    SIGNED = "SIGNED"


class ContractInitRequest(BaseModel):
    chatRoomId: str


class ContractUpdateRequest(BaseModel):
    contractData: dict[str, object]


class ContractResponse(BaseModel):
    id: str
    chatRoomId: str
    userAId: str
    userBId: str
    nicknameA: str | None = None
    nicknameB: str | None = None
    status: ContractStatus
    contractData: dict[str, object]
    signatureA: bool
    signatureB: bool
    signedAt: datetime | None
    endSemesterA: bool = False
    endSemesterB: bool = False

    class Config:
        from_attributes = True
